### PR TITLE
Battle

### PR DESCRIPTION
**Pull request by antu3199**

### CODE REVIEW

File: Assets/_Scripts/BattleController.cs [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+    List<Player> _players;
+    Player _heroPlayer;
+
+
+    bool startActions = false;
+
+
+    void Start() {
+        _players = GameManager.Instance.party.GetPlayersInPosition();
```
Comment (BenjixD) [inline Assets/_Scripts/BattleController.cs:36]: Do players ever get updated when party changes?

File: Assets/_Scripts/BattleController.cs [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+        _heroPlayer = _players[0];
+
+        Debug.Log("Players: " + _players.Count);
+        for (int i = 0; i < _players.Count; i++) {
+            Debug.Log(_players[i].Name);
+        }
+
+        // TODO: Waves
+        GameManager.Instance.GenerateEnemyList();
```
Comment (BenjixD) [inline Assets/_Scripts/BattleController.cs:45]: We could store enemies like players every turn. That way we only need to interact with game manager at the beginning of the turn.

File: Assets/_Scripts/BattleController.cs [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+        this.OnPlayerTurnStart();
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
```
Comment (BenjixD) [inline Assets/_Scripts/BattleController.cs:51]: As a suggestion, consider using recursive coroutines for the two turn phases.

File: Assets/_Scripts/BattleController.cs [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+        foreach (var player in _players) {
+            player.hasSetCommand = false;
+
+            if (player.modifiers.Contains("defend")) {
+                player.modifiers.Remove("defend");
+            }
+        }
+    }
+
+    public void DoAction(FightingEntity a, FightingEntity b) {
```
Comment (BenjixD) [inline Assets/_Scripts/BattleController.cs:182]: I would suggest we actually make executing actions a class in the future. This allows us for slightly more complex involvement.

File: Assets/_Scripts/BattleController.cs (hunk 1 above)
Comment (antu3199) [reply Assets/_Scripts/BattleController.cs:44]: I was thinking about changing that later. Maybe instead I'll make a helper function that calls GameManager.Instance.party.GetPlayersInPosition() each time. It won't be as efficient, but the size of the list is 4 so it shouldn't matter.

**Pull request merged**